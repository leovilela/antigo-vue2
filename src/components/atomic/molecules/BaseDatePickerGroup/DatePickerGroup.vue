<template>
  <div :class="formClass">
    <label for="" v-if="label">{{label}}</label>
    <date-picker :lang="'pt-br'" :value="value" :format="'DD/MM/YYYY'" v-bind="$attrs" @input="$emit('input', $event)" @change="$emit('change', $event)" />
  </div>
</template>

<script>
import DatePicker from "vue2-datepicker";

export default {
  name: "base-datepickergroup",
  components: {
    DatePicker,
  },
  props: {
    formClass: {
      type: Object,
      default: () => {},
    },
    label: {
      type: String,
      default: null,
    },
    value: {
      type: [String, Boolean, Date, Object],
      default: "",
    },
  },
};
</script>

<style lang="less" scoped>
@import "~assets/vars.less";
label {
  font-size: @ftSizeSmall;
  font-family: @mainFontRegular;
}
</style>
