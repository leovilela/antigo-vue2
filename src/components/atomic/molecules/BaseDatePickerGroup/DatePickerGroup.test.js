import { shallowMount } from "@vue/test-utils";
import DatePickerGroup from "../BaseDatePickerGroup/DatePickerGroup";

const factory = () => shallowMount(DatePickerGroup, {
  stubs: ["date-picker"],
});

describe("DatePickerGroup", () => {
  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
