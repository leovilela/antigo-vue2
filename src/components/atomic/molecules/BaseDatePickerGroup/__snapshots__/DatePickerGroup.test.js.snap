// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatePickerGroup renders properly 1`] = `
<div>
  <!---->
  <date-picker-stub value="" valuetype="date" lang="pt-br" format="DD/MM/YYYY" type="date" rangeseparator="~" confirmtext="OK" editable="true" clearable="true" shortcuts="true" inputname="date" inputclass="mx-input"></date-picker-stub>
</div>
`;
